Client/Frontend/Home/FirefoxHomeRecentlySavedViewModel.swift

```diff
@@ -3,16 +3,94 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0
 
 import Foundation
+import Storage
+
+protocol RecentlySavedItem {
+    var title: String { get }
+    var url: String { get }
+}
+
+extension ReadingListItem: RecentlySavedItem { }
+extension BookmarkItem: RecentlySavedItem { }
 
-// TODO: Move more items from FxHomeRecentlySavedCollectionCell into here.
 class FirefoxHomeRecentlySavedViewModel {
     
     // MARK: - Properties
-    
-    var recentItems = [RecentlySavedItem]()
+
     var isZeroSearch: Bool
 
-    init(isZeroSearch: Bool) {
+    private let profile: Profile
+    private lazy var siteImageHelper = SiteImageHelper(profile: profile)
+    private var readingListItems = [ReadingListItem]()
+    private var recentBookmarks = [BookmarkItem]()
+
+    init(isZeroSearch: Bool, profile: Profile) {
         self.isZeroSearch = isZeroSearch
+        self.profile = profile
+    }
+
+    var recentItems: [RecentlySavedItem] {
+        var items = [RecentlySavedItem]()
+        items.append(contentsOf: recentBookmarks)
+        items.append(contentsOf: readingListItems)
+
+        return items
+    }
+
+    // Whether the section is enabled or not
+    var isEnabled: Bool {
```

Comment:
This is misleading because of the vocabulary we use in other places. Let's rename this to something like: `hasData` or something. The section being enabled or not should also include whether or not the section is actually enabled for the build/user.

Reply:
Will do!